src/modules/ToDoList/ToDoList.js

```diff
@@ -0,0 +1,85 @@
+import React from "react";
+
+import '../../styles/ToDoListStyles/ToDoList.css'
+import ToDoTask from "./ToDoTask";
+
+import {connect} from 'react-redux'
+import {
+    getDeletedTasks,
+    getModifiedTasks,
+    getNewTasks,
+    getStoreTasks,
+    isFetching,
+    listHasUnsavedChanges
+} from "./_selectors";
+import {addTask, deleteTasksFromServer, postTasksToServer, putTasksToServer} from "./_actionsDefinitions";
+
+
+class ToDoList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {inputValue: ''};
+    }
+
+    onInputChange = (event) => this.setState({ inputValue: event.target.value});
+
+    addTaskOnAddButton = (event) => {
+        event.preventDefault();
+        this.props.addTask(this.state.inputValue);
+        this.setState({inputValue: ""});
+    };
+
+    saveChanges = () => {
+        this.props.postTasksToServer(this.props.newTasks);
```

Comment:
add a condition to call it only if new tasks are present. There is no need to dispatch action if no new data comes
use the same logic for the actions below